include/libtorrent/packet_pool.hpp

```diff
@@ -78,78 +78,82 @@ namespace libtorrent
 		std::uint8_t buf[1];
 	};
 
+
+	struct packet_deleter
+	{
+		void operator()(packet* handle) const
+		{
+			std::free(handle);
+		}
+	};
+
+	using packet_ptr = std::unique_ptr<packet, packet_deleter>;
+
+	template<std::size_t ALLOCATE_SIZE>
+	struct TORRENT_EXTRA_EXPORT packet_slab : private aux::vector<packet_ptr, std::size_t>
+	{
+		static const std::size_t allocate_size{ ALLOCATE_SIZE };
+
+		packet_slab(std::size_t limit) : m_limit(limit) { reserve(m_limit); }
+
+		void try_free(packet_ptr &p)
+		{
+			if (size() < m_limit)
+				push_back(std::move(p));
+		}
+
+		void try_alloc(packet_ptr &p)
+		{
+			if (empty())
+				p.reset(static_cast<packet*>(std::malloc(sizeof(packet) + allocate_size)));
```

Comment:
would you mind calling the ``packet`` constructor here too?
``new (p.get()) packet();``